doc/Language/typesystem.pod6

```diff
@@ -258,6 +258,49 @@ you must use private attribute notation instead.
     C.new.say; C.new('answer').say;
     # OUTPUT: «C.new(attr => 42)␤
     #          C.new(attr => "answer")␤»
+    
+=head4 submethod TWEAK
+
+The L<submethod|/type/Submethod> C<TWEAK> is (indirectly) called by
+L<.bless|/type/Mu#method_bless>. It
+is meant to set private and public attributes of a class after the BUILD submethod has been called. The default
+constructor L<.new|/type/Mu#method_new> defined in C<Mu> is the method that
+invokes it. Given that public accessor methods are not available in C<TWEAK>,
+you must use private attribute notation instead.
+
+The difference between BUILD and TWEAK is that not all the attributes in an object have been built.
+In addition, when there is inheritance the BUILD submethods of the child classes are called after
+the BUILD submethods of the parent classes. The TWEAK submethods are called after all the BUILD 
+submethods.
+
+Both C<BUILD> and C<TWEAK> receive all named
+attributes passed into C<.bless>
+
+    class C {
+        has $.m = 'nothing';
+        has $.attr = 42;
+        submethod BUILD {
+            # this will fail because $!attr has not been given its default value yet
+            $!message = 'A very special number' if $!attr == 42;
+        }
+    };
+    class D {
+        has $.m = 'nothing';
+        has $.attr = 42;
+        submethod TWEAK {
+            $!message = 'A very special number' if $!attr == 42;
+        }
+    };
+
+    C.new.m.say;
+    # OUTPUT: 
+    # Use of uninitialized value of type Any in numeric context
+    # in submethod BUILD at <unknown file> line 1
+    # in block <unit> at <unknown file> line 1
+    #
+    # nothing
+    D.new.m.say;
+    # OUTPUT: A very special number
 
```

Comment:
This example is OK because it shows when to use attributes and when they are not available; but still, I'm not too convinced about the rest of the explanation.

Reply:
The example shows exactly why TWEAK is needed. Naturally its a bit simple, but a more complex situation needs more code. 

Re #1808 - I can add that no problem.
Re #1796 - I think my text resolves the issue. It explains what TWEAK is, and why it is needed in addition to BUILD.

Reply:
I did change the TWEAK text to add what AlexDaniels wanted. But it has got separated from this patch.

Reply:
This is a branch, the other is another. If we merge the other, it will conflict with this one. Please close the other one, or just make an atomic commit with the small change that's requested (I don't know where, BTW)

Reply:
i'm trying to resolve each matter